ls1intum/Artemis
# `Lectures`: Automatically set name for attachments

It is cumbersome to add the title manually if it should be similar to the file name, therefore Artemis now adds it automatically for file attachments and attachment units in case it is not yet specified.
This PR also improves the involved Typescript types to avoid `any`.

### Checklist
#### General



### Steps for Testing

Prerequisites:
- 1 Instructor

1. Create a lecture
2. Add a lecture attachment by choosing a file
3. The name should be filled automatically without the file extension
4. Choose a different file --> the name should not change
5. Add an attachment unit by choosing a file
6. The name should be filled automatically without the file extension
7. Choose a different file --> the name should not change


### Review Progress
#### Code Review
#### Manual Tests
